python/hackathon_app/sheets.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime
+
+from google.oauth2 import service_account  # type: ignore
+from googleapiclient import discovery  # type: ignore
+
+from typing import Sequence, List, Dict
+
+
+class Sheets:
+    """An API for manipulating the Google Sheet containing hackathon data."""
+
+    def __init__(self, spreadsheet_id):
+        scopes = [
+            "https://www.googleapis.com/auth/drive",
+            "https://www.googleapis.com/auth/drive.file",
+            "https://www.googleapis.com/auth/spreadsheets",
+        ]
+
+        credentials = service_account.Credentials.from_service_account_file(
+            "jaxdata-a93316beb3a0.json", scopes=scopes
+        )
+
+        service = discovery.build("sheets", "v4", credentials=credentials)
+        self.spreadsheet = service.spreadsheets().values()
+        self.id = spreadsheet_id
+
+    def get_hackathons(self):
+        """Get names of active hackathons."""
+        hackathons_table = self.spreadsheet.get(
+            spreadsheetId=self.id, range="hackathons!A1:end"
+        ).execute()
+        data = hackathons_table["values"][1:]
+        date_index = hackathons_table["values"][0].index("date")
+        hackathons = []
+        for row in data:
+            if datetime.strptime(row[date_index], "%m/%d/%Y") >= datetime.now():
+                hackathons.append(row[1])
+        return hackathons
+
+    def register_user(
+        self,
+        hackathon: str,
+        first_name: str,
+        last_name: str,
+        email: str,
+        company: str,
+        country: str,
+        tshirt_size: str,
+    ):
+        """Register user to a hackathon"""
+        if not self._is_created(email):
+            self.create_user(
+                first_name, last_name, email, company, country, tshirt_size
+            )
+        self._register_user(email, hackathon)
```

Comment:
I see that this functionality is missing - I think it can wait for the next round though.